sparse_ho/algo/forward.py

```diff
@@ -104,14 +104,14 @@ def get_beta_jac_iterdiff(
         alphas = np.ones(n_features) * alpha
     ############################################
     # warm start for beta
-    beta, r = model._init_beta_r(X, y, mask0, dense0)
+    beta, residuals = model._init_beta_residuals(X, y, mask0, dense0)
```

Comment:
since you call it beta and not betas I would suggest you call it residual (no s)